__tests__/services/users.service.spec.ts

```diff
@@ -1,32 +1,87 @@
 import { Container } from 'typedi';
 import { genSaltSync, hashSync } from 'bcrypt';
 import { UsersService } from '@services/users.service';
+import { factory } from 'typeorm-seeding';
+import { User } from '@entities/user.entity';
+import { getRepository, Repository } from 'typeorm';
+import { mockUpdateResult } from '../utils/mocks';
+import { HashInvalidError } from '@exception/users/hashinvalid.error';
+import { HashExpiredError } from '@exception/users/hashexpired.error';
 
 let usersService: UsersService;
+let user: User;
+let userRepository: Repository<User>;
 
-beforeAll(async () => {
-  usersService = Container.get(UsersService);
-});
-
-describe('compare password', () => {
-  let userPassword: string;
+describe('UsersService', () => {
+  beforeAll(async () => {
```

Comment:
Take in mind that declaring async things that are not needed to be async (like this, you need to ensure this is executed before everything else) could lead to unexpected behaviors of the testing suite.